enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,179 @@
+---
+title: enable-baremetal-on-other-platforms
+authors:
+  - "@asalkeld"
+  - "@sadasu"
+reviewers:
+  - TBD
+  - "@hardy"
+  - "@romfreiman"
+  - "@dhellman"
+approvers:
+  - TBD
+creation-date: 2021-08-20
+last-updated: 2021-08-20
+status: implementable
+see-also:
+  - "/enhancements/baremetal-provisiong-config.md"
+replaces:
+superseded-by:
+---
+
+# Enable Baremetal on other Platforms to support ZTP
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+BMO is only available when deploying an OpenShift cluster with the baremetal
+platform (via the IPI or AI workflow). Having the ability to manage baremetal nodes from
+clusters without requiring the cluster to be on baremetal would be beneficial
+to customers.
+
+## Motivation
+
+An initial driver of this feature are the centralized host management use cases
+in edge topologies, which without this feature, is restricted to having the
+central OpenShift cluster deployed on bare metal.
+
+### Goals
+
+The specific goals of this proposal are to:
+
+Support the ZTP use case by partially enabling cluster-baremetal-operator on the following platforms:
+- None
+- OpenStack
+- vSphere
+
+We will be successful when:
+
+ZTP can deploy clusters using the above platforms.
+
+### Non-Goals
+
+Allow CBO to be fully enabled on all platforms.
+
+Note that the scope of this epic is only enablement of the Metal3
+BaremetalHost API and associated controller/services - the edge ZTP flow
+currently interacts directly via this API without any Machine API integration.
+
+## Proposal
+
+Currently cluster-baremetal-operator checks the platform and if it is not
+baremetal it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow cluster-baremetal-operator to be enabled for the
+ZTP use case which is Platforms: None, OpenStack or vSphere.
+
+Further (to restrict the testing matrix) the allowed configuration options
+of the Provisioning CR will be restricted to exactly those required by ZTP.
+
+*Only spec.provisioningNetwork=Disabled mode will be accepted in the Provisioning CR.*
```

Comment:
What will the CBO do if another network mode is used?

Reply:
CBO has a webhook, so we can just fail the validation in the validating webhook.

Reply:
That won't help in a case where a customer has created a Provisioning resource before upgrading to the version of CBO that has the validation, right?

Reply:
Yes, correct. I'll have to remove this check here https://github.com/openshift/cluster-baremetal-operator/blob/master/controllers/provisioning_controller.go#L257-L270 so that that validation always runs in the Reconcile.
Kinda a corner case..